packages/core/src/declareAtom.ts

```diff
@@ -160,66 +140,64 @@ export function declareAtom<State>(
   return atom
 }
 
-export function getState<T>(state: State, atom: Atom<T>): T | undefined {
+export function getState<T>(state: State, atom: Atom<T>): T {
   return state[atom[TREE].id]
 }
 
-// @ts-ignore
-export declare function map<T, _T = unknown>(
+export function map<T, _T = unknown>(
```

Comment:
Type inference for the map is broking
![proof](https://user-images.githubusercontent.com/27290320/65603737-e32a1400-dfae-11e9-870c-aa51150d43cc.png)